Release Gateway 3.4.2.0 (#6490)
Kong/docs.konghq.com: app/gateway/changelog.md

For product versions that have reached the end of sunset support, see the [changelog archives](https://legacy-gateway--kongdocs.netlify.app/enterprise/changelog/).

## 3.4.2.0 
**Release date** 2023/11/08

### Features
#### Enterprise

* License management:
  * Implemented a new grace period that lasts 30 days from the Kong Enterprise license expiration date. 
    During the grace period all open source functionality will be available, and
    Enterprise functionality will be set to read-only mode.
  * Added support for counters such as routes, plugins, licenses, and deployment information to the license report.
  * Added a checksum to the output of the license endpoint.
### Fixes
#### Core
* DNS
  * Fixed an issue with the DNS client was not adhering to configured timeouts in a predictable manner. [#11386](https://github.com/Kong/kong/pull/11386)

>   * Fixed an issue that caused the DNS client to resolve incorrectly during transient network and DNS server failures. [#11386](https://github.com/Kong/kong/pull/11386)

Reviewer: These are the same issue with two different entries (see the link to the PR). You should combine them into one.

* The default value of the [`dns_no_sync`](/gateway/3.5.x/reference/configuration/#dns_no_sync) option has been changed to `on`.
[#11871](https://github.com/kong/kong/pull/11871).
#### Plugins

> * [**OpenTelemetry**](/hub/kong-inc/opentelemetry/) (`opentelemetry`): Fixed an issue that resulted in traces with invalid parent IDs when balancer instrumentation was enabled. [#11830](https://github.com/Kong/kong/pull/11830)

Reviewer: ```suggestion
* [**OpenTelemetry**](/hub/kong-inc/opentelemetry/) (`opentelemetry`): 
  * Fixed an issue that resulted in traces with invalid parent IDs when balancer instrumentation was enabled. [#11830](https://github.com/Kong/kong/pull/11830)
```
I know this formatting seems excessive, but it's consistent with the rest of the changelog.

> * [**OpenID Connect**](/hub/kong-inc/openid-connect/) (`openid-connect`): Fixed an issue with `using_pseudo_issuer`, where once propogated it was not used. [#6981](https://github.com/Kong/kong-ee/pull/6981)

Reviewer: ```suggestion
* [**OpenID Connect**](/hub/kong-inc/openid-connect/) (`openid-connect`)
  * Fixed an issue with `using_pseudo_issuer`, where it was not used after it was propagated.
```
We can't add links to kong-ee in the changelog.

Not sure about this rephrase, you're welcome to adjust.

> #### Kong Manager 

Reviewer: The Kong Manager section usually comes before Plugins.

* Fixed an issue where some services were not showing the exact name or ID while configuring a route. 

### Dependencies
#### Enterprise

> * bump OpenSSL to 3.1.4
> * Add troubleshooting tools to container images 
> 
> 
> * Bumped `ngx_wasm_module` version to prerelease-0.1.1 

Reviewer: ```suggestion
* Bumped OpenSSL to 3.1.4
* Added troubleshooting tools to container images
* Bumped `ngx_wasm_module` version to prerelease-0.1.1 
```

> * Bumped `ngx_wasm_module` version to prerelease-0.1.1 

Reviewer: Bumped from what? Usually we do "bumped x from <some-version> to <new-version>".

  * Bumped `curl` from 8.3.0 to 8.4.0
  * Bumped `nghttp2` from 1.56.0 to 1.57.0

## 3.4.2.0 
**Release date** 2023/11/08

### Features
#### Enterprise

* License management:
  * Implemented a new grace period that lasts 30 days from the Kong Enterprise license expiration date. 
    During the grace period all open source functionality will be available, and
    Enterprise functionality will be set to read-only mode.
  * Added support for counters such as routes, plugins, licenses, and deployment information to the license report.
  * Added a checksum to the output of the license endpoint.
### Fixes
#### Core
* Fixed an issue with the DNS client was not adhering to configured timeouts in a predictable manner. Also fixed a related issue that cause the DNS client to resolve incorrectly during transient network and DNS server failures. [#11386](https://github.com/Kong/kong/pull/11386)
* The default value of the [`dns_no_sync`](/gateway/3.5.x/reference/configuration/#dns_no_sync) option has been changed to `on`.

> [#11871](https://github.com/kong/kong/pull/11871).

Reviewer 2: ```suggestion
[#11871](https://github.com/kong/kong/pull/11871).
* Removed a confusing log from the Redis rate limiting tool.
```

  * Bumped `curl` from 8.3.0 to 8.4.0
  * Bumped `nghttp2` from 1.56.0 to 1.57.0

## 3.4.2.0 
**Release date** 2023/11/08

### Features
#### Enterprise

* License management:
  * Implemented a new grace period that lasts 30 days from the Kong Enterprise license expiration date. 
    During the grace period all open source functionality will be available, and
    Enterprise functionality will be set to read-only mode.
  * Added support for counters such as routes, plugins, licenses, and deployment information to the license report.
  * Added a checksum to the output of the license endpoint.
### Fixes
#### Core

> * DNS
>   * Fixed an issue with the DNS client was not adhering to configured timeouts in a predictable manner. Also fixed a related issue that cause the DNS client to resolve incorrectly during transient network and DNS server failures. [#11386](https://github.com/Kong/kong/pull/11386)

Reviewer: ```suggestion
* Fixed an issue with the DNS client was not adhering to configured timeouts in a predictable manner. Also fixed a related issue that cause the DNS client to resolve incorrectly during transient network and DNS server failures. [#11386](https://github.com/Kong/kong/pull/11386)
```

  * Bumped `curl` from 8.3.0 to 8.4.0
  * Bumped `nghttp2` from 1.56.0 to 1.57.0

## 3.4.2.0 

> **Release date** 2023/11/09

Reviewer 2: ```suggestion
**Release date** 2023/11/10
```
